domain/schema/model/sql/0018-machine.sql

```diff
@@ -0,0 +1,74 @@
+CREATE TABLE machine (
+    uuid              TEXT PRIMARY KEY,
+    machine_id        TEXT NOT NULL,
+    net_node_uuid     TEXT NOT NULL,
+    life_id           INT NOT NULL,
+    base              TEXT,
+    nonce             TEXT,
+    password_hash     TEXT,
+    clean             BOOLEAN,
+    force_destroyed   BOOLEAN,
+    placement         TEXT,
+    agent_started_at  DATETIME,
+    hostname          TEXT,
+    is_controller     BOOLEAN,
+    CONSTRAINT        fk_machine_net_node
+        FOREIGN KEY   (net_node_uuid)
+        REFERENCES    net_node(uuid),
+    CONSTRAINT        fk_machine_life
+        FOREIGN KEY   (life_id)
+        REFERENCES    life(id)
+);
+
+CREATE UNIQUE INDEX idx_machine_id
+ON machine (machine_id);
+
+CREATE UNIQUE INDEX idx_machine_net_node
+ON machine (net_node_uuid);
+
+CREATE TABLE machine_constraint (
+    machine_uuid     TEXT PRIMARY KEY,
+    constraint_uuid  TEXT NOT NULL,
+    CONSTRAINT       fk_machine_constraint_machine
+        FOREIGN KEY  (machine_uuid)
+        REFERENCES   machine(uuid),
+    CONSTRAINT       fk_machine_constraint_constraint
+        FOREIGN KEY  (constraint_uuid)
+        REFERENCES   "constraint"(uuid)
+);
+
+CREATE TABLE machine_tool (
+    machine_uuid     TEXT NOT NULL,
+    tool_url         TEXT NOT NULL,
+    tool_version     TEXT NOT NULL,
+    tool_sha256      TEXT NOT NULL,
+    tool_size        INT NOT NULL,
+    CONSTRAINT       fk_machine_principal_machine
+        FOREIGN KEY  (machine_uuid)
+        REFERENCES   machine(uuid),
+    PRIMARY KEY (machine_uuid, tool_url)
+);
+
+CREATE TABLE machine_volume (
```

Comment:
I don't know if we need these. I believe they will be implied by being against units, but we can delete land this and delete later. @wallyworld to confirm.